### PR TITLE
dts: msm8916: coolpad-coolpady80d (WIP display support)

### PR DESCRIPTION
**Pull request by nphuracm**
These are created mainly for use with lk1st due to the device's quirky bootloader.
I could not get the display to work at the moment.

### CODE REVIEW

File: dts/msm8916/msm8916-coolpad-coolpady80d.dts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+#include <skeleton.dtsi>
+
+/ {
+	#address-cells = <0x02>;
+	#size-cells = <0x02>;
+	qcom,msm-id = <0xce 0x00>;
+	qcom,board-id = <0x08 0x00>;
+	model = "Coolpad Y80D";
+	compatible = "coolpad,coolpady80d\0qcom,msm8916-mtp\0qcom,msm8916\0lk2nd,device\0qcom,mtp";
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8916/msm8916-coolpad-coolpady80d.dts:11]:
```suggestion
	compatible = "coolpad,coolpady80d", "qcom,msm8916", "lk2nd,device";
```

File: lk2nd/panel/generated/panels.h [hunk 2]
```diff
@@ -10,5 +10,6 @@
 #include "lk_panel_r69431_720p_video.h"
 #include "lk_panel_tianma_nt35521_5p5_720p_video.h"
 #include "lk_panel_yushun_nt35520_720p_cmd.h"
+#include "lk_panel_otm9605a_lide_qhd_video_550.h"
```
Comment (wonderfulShrineMaidenOfParadise) [inline lk2nd/panel/generated/panels.h:13]: The headers should be sorted alphabetically.

File: dts/msm8916/msm8916-coolpad-coolpady80d.dts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/dts-v1/;
+
+#include <skeleton.dtsi>
+
+/ {
+	#address-cells = <0x02>;
+	#size-cells = <0x02>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8916/msm8916-coolpad-coolpady80d.dts:7]:
```suggestion
```
It's already included from `skeleton.dtsi`.

File: dts/msm8916/msm8916-coolpad-coolpady80d.dts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	qcom,msm-id = <0xce 0x00>;
+	qcom,board-id = <0x08 0x00>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8916/msm8916-coolpad-coolpady80d.dts:9]:
```suggestion
	qcom,msm-id = <206 0>;
	qcom,board-id = <8 0>;
```

By the way it seems to be msm8916-mtp and would be better to merge this dts into `msm8916-mtp.dts`

File: lk2nd/panel/generated/lk_panel_otm9605a_lide_qhd_video_550.h [hunk 5]
```diff
@@ -0,0 +1,802 @@
...
+	0x02, 0x00, 0x39, 0xc0, 0x55, 0x01, 0xff, 0xff
+};
+static char otm9605a_lide_qhd_video_550_on_cmd_143[] = {
+	0x11, 0x00, 0x05, 0x80
+};
+static char otm9605a_lide_qhd_video_550_on_cmd_144[] = {
+	0x29, 0x00, 0x05, 0x80
+};
+static char otm9605a_lide_qhd_video_550_on_cmd_145[] = {
+	0x51, 0xfa, 0x15, 0x80 /* https://github.com/msm8916-mainline/lk2nd/commit/3d8b68ddef17afeaba7b34180489370598cdaaea */
```
Comment (wonderfulShrineMaidenOfParadise) [inline lk2nd/panel/generated/lk_panel_otm9605a_lide_qhd_video_550.h:566]:
```suggestion
	0x51, 0xff, 0x15, 0x80 /* https://github.com/msm8916-mainline/lk2nd/commit/3d8b68ddef17afeaba7b34180489370598cdaaea */
```

The maximum is 255. See `otm9605a_lide_qhd_video_550_backlight`

File: dts/msm8916/msm8916-coolpad-coolpady80d.dts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+*/
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+
+/ {
+
+	qcom,msm-id = <206 0>;
+	qcom,board-id = <8 0>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8916/msm8916-coolpad-coolpady80d.dts:29]: https://github.com/msm8916-mainline/lk2nd/blob/606020a090f38b8bb28acc82106f64f6b5587f96/dts/msm8916/msm8916-mtp.dts#L11-L12

Duplicated id due to recent changes on branch `master`. This device tree is supposed to be merged into `msm8916-mtp` now.

Comment (nphuracm) [reply dts/msm8916/msm8916-coolpad-coolpady80d.dts:29]: So I am supposed to deprecate this file and instead add panel code in `msm8916-mtp.dts`?

Comment (TravMurav) [reply dts/msm8916/msm8916-coolpad-coolpady80d.dts:29]: In fact, was supposed from the start, since 206/8 is the id for MTP

**Comment by nphuracm** [comment]
Merged to msm8916-mtp.dts.

File: dts/msm8916/msm8916-mtp.dts [hunk 7]
```diff
@@ -39,4 +39,17 @@
 		compatible = "asus,z010d", "qcom,msm8916", "lk2nd,device";
 		lk2nd,match-cmdline = "* mdss_mdp.panel=1:dsi:0:qcom,mdss_dsi_r69339_720p_video:1:none";
 	};
+
+	coolpad-coolpady80d {
```
Comment (wonderfulShrineMaidenOfParadise) [inline dts/msm8916/msm8916-mtp.dts:43]: If this phone is quirky, you need to leave comments about workarounds here.

**Comment by nphuracm** [comment]
Obsoleting this for a while. I'll open another PR when I decide to redo this. See: [https://github.com/msm8916-mainline/linux/pull/300](my other PR).

**Pull request closed without merging**